ui/pages/settings/contact-list-tab/contact-networks.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import { NetworkConfiguration } from '@metamask/network-controller';
+import {
+  Display,
+  FlexDirection,
+  BlockSize,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+
+import {
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  Modal,
+  Box,
+  Text,
+  AvatarNetworkSize,
+} from '../../../components/component-library';
+
+import { getNetworkConfigurationsByChainId } from '../../../../shared/modules/selectors/networks';
+import { TEST_CHAINS } from '../../../../shared/constants/network';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { NETWORK_TO_SHORT_NETWORK_NAME_MAP } from '../../../../shared/constants/bridge';
+import { NetworkListItem } from '../../../components/multichain';
+import { getImageForChainId } from '../../../selectors/multichain';
+
+export const ContactNetworks = ({
+  isOpen,
+  onClose,
+  selectedChainId,
+  onSelect,
+}: {
+  isOpen: boolean;
+  onClose: () => void;
+  selectedChainId?: string;
+  onSelect?: (chainId: string) => void;
+}) => {
+  const t = useI18nContext();
+
+  const networkConfigurations = useSelector(getNetworkConfigurationsByChainId);
+  const [nonTestNetworks, testNetworks] = useMemo(
+    () =>
+      Object.entries(networkConfigurations).reduce(
+        ([nonTest, test], [chainId, config]) => {
+          const isTest = (TEST_CHAINS as string[]).includes(chainId);
+          (isTest ? test : nonTest).push(config);
+          return [nonTest, test];
+        },
+        [[] as NetworkConfiguration[], [] as NetworkConfiguration[]],
+      ),
+    [networkConfigurations],
+  );
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onClose={onClose}
+      className="multichain-asset-picker__network-modal"
+    >
+      <ModalOverlay />
+      <ModalContent modalDialogProps={{ padding: 0 }}>
+        <ModalHeader onBack={onClose} onClose={onClose}>
+          {t('bridgeSelectNetwork')}
+        </ModalHeader>
+        <Box
+          className="multichain-asset-picker__network-list"
+          display={Display.Flex}
+        >
+          <Box
+            display={Display.Flex}
+            flexDirection={FlexDirection.Column}
+            width={BlockSize.Full}
+          >
+            {nonTestNetworks.map(({ name, chainId }) => (
+              <NetworkListItem
```

Comment:
Is this block the same in both test and nontest networks?  Should this be a helper function to create the element?

Reply:
Done https://github.com/MetaMask/metamask-extension/pull/31488/commits/60b31804f63afc480c5d02c920b48d817f6f211b